src/node_worker.cc

```diff
@@ -661,6 +663,9 @@ void Worker::StartThread(const FunctionCallbackInfo<Value>& args) {
   ASSIGN_OR_RETURN_UNWRAP(&w, args.This());
   Mutex::ScopedLock lock(w->mutex_);
 
+  // Store a reference to the worker so that it can be used
+  // for inter thread communication
+  workers[w->thread_id_.id] = w;
```

Comment:
Is the entry in the map removed if starting the thread via `uv_thread_create_ex` fails?

Reply:
I changed the code. Now it will only be assigned if the call succeeded.